app/forms/nominate_induction_tutor_form.rb

```diff
@@ -5,70 +5,94 @@ class NominateInductionTutorForm
   include ActiveRecord::AttributeAssignment
   include ActiveModel::Serialization
 
-  attr_accessor :full_name, :email, :token, :school_id, :user_id
+  attr_accessor :email, :full_name, :school, :token
 
-  validates :full_name, presence: true, on: :full_name
-  validates :email, presence: true, notify_email: true, on: :email
-  validate :different_name, on: :full_name
-  validate :email_is_not_in_use, on: :email
-  validate :name_matches, on: :email
+  # Validations
+  validates :email, presence: true, notify_email: true, on: %i[email]
+  validates :full_name, presence: true, on: %i[full_name email]
+  validates :school, presence: true, on: %i[email]
+  validate :not_current_sit, on: %i[email]
+  validate :name_matches, on: %i[email]
 
+  # Instance Methods
   def attributes
-    {
-      full_name:,
-      email:,
-      token:,
-    }
+    { email:, full_name:, token: }
   end
 
-  def school
-    if school_id
-      School.find school_id
-    else
-      NominationEmail.find_by(token:).school
+  def save!
+    raise errors.full_messages unless valid?(:email)
+
+    ActiveRecord::Base.transaction do
+      remove_current_sits
+      ensure_user_persisted
+      add_school_to_sit
+      inform_user
     end
+
+    true
   end
 
-  def existing_name
-    user_scope = user_id.present? ? User.induction_coordinators.where.not(id: user_id) : User.induction_coordinators
-    existing_user = user_scope.find_by(email:)
-    return if existing_user.blank?
+private
 
-    existing_user.full_name
+  def add_school_to_sit
+    sit_profile.schools << school
   end
 
-  def name_different?
-    existing_name.present? && existing_name != full_name
+  def different_name?
+    user.full_name != full_name
   end
 
-  def email_already_taken?
-    Identity.find_user_by(email:)
+  def ensure_user_persisted
+    user.save! unless user.persisted?
   end
 
-private
+  def inform_user
+    SchoolMailer.with(sit_profile:, school:, start_url:, step_by_step_url:)
+                .nomination_confirmation_email.deliver_later
+  end
 
-  def email_is_not_in_use
-    if email_already_taken?
-      other_school = school_using_this_email(email)
-      if other_school.present?
-        errors.add(:email, "The email address #{email} is already used by a user at #{other_school.name}")
-      else
-        errors.add(:email, "The email address #{email} is already in use")
-      end
+  def name_matches
+    if different_name?
+      errors.add(:full_name, "A user with a different name (#{user.full_name}) has already been registered with this email address. Please, change the name or email you have entered.")
```

Comment:
```suggestion
      errors.add(:full_name, "A user with a different name (#{user.full_name}) has already been registered with this email address. Change the name or email address you entered.")
```

Reply:
I have made my own commit for this suggestion because there were other replicas of this messages to update as well.